sdk/src/main/java/io/customer/sdk/data/store/FileStorage.kt

```diff
@@ -0,0 +1,88 @@
+package io.customer.sdk.data.store
+
+import android.content.Context
+import java.io.File
+
+/*
+ Save data to a file on the device file system.
+
+ Responsibilities:
+ * Be able to mock so we can use in unit tests without using the real file system
+ * Be the 1 source of truth for where certain types of files are stored.
+   Making code less prone to errors and typos for file paths.
+
+ Way that files are stored in this class:
+ ```
+ Internal storage (since we don't need to ask for permissions and its private to the app)
+ io.customer/   // Create a directory to separate SDK files from other app files.
+   <site-id>/   // sandbox all files in the SDK to it's site-id.
+     queue/
+       inventory.json
+       tasks/
+         <task-id>.json
+ ```
+
+ Notice that we are using the <site id> as a way to isolate files from each other.
+ The file tree remains the same for all site ids.
+ */
+class FileStorage internal constructor(
```

Comment:
How do you feel about using `SQLite` instead of a file-based storage system? We will still not be adding any additional dependency, more command on how we interact with it. If we have to sort the tasks, add in priority it would be alot easier. What do you think?

Reply:
In the past, I have [built a background queue that uses SQLite](https://github.com/levibostian/wendy-android). I encountered some pains with releasing a 3rd party library that uses SQLite. I will admit I encountered more pains with [the iOS version](https://github.com/levibostian/wendy-ios). 

When building the background queue for our iOS SDK, I tried the file system approach instead of a database as an experiment. It went very well. I didn't encounter the same issues that using a database gave me. With that in mind, I suggest that we stick with using the file system knowing that if we ever encounter problems with this approach, we can migrate to a database in the future.

Reply:
Sounds good we can start off with a file system, but if we want it to be flexible, should we then probably have a `QueueStorage` interface, with `FileStorage` as its implementation? 

```kotlin
class QueueStorageImpl internal constructor(
    private val siteId: String,
    private val fileStorage: QueueStorage,
    private val jsonAdapter: JsonAdapter
) ```